src/Network/Mattermost/WebSocket/Types.hs

```diff
@@ -235,6 +236,10 @@ instance FromJSON WEData where
     wepPreferences <- case wepPreferencesRaw of
       Just str -> fromValueString str
       Nothing  -> return Nothing
+    wepChannelMemberRaw <- o .:? "channelMember"
+    wepChannelMember <- case wepChannelMemberRaw of
```

Comment:
This bind/`case` pair can be collapsed to `wepChannelMember <- fmap fromValueString <$> o .:? "channelMember"`.

Reply:
It ended up being slightly different due to the type of `fromValueString`. I also applied the same simplification to the other bindings in that `do` block.